examples/09-ai/05-manual-execution/src/App.tsx

```diff
@@ -0,0 +1,206 @@
+      },
+      {
+        type: "paragraph",
+        content:
+          "Open source software refers to computer programs whose source code is made available to the public, allowing anyone to view, modify, and distribute the code. This model stands in contrast to proprietary software, where the source code is kept secret and only the original creators have the right to make changes. Open projects are developed collaboratively, often by communities of developers from around the world, and are typically distributed under licenses that promote sharing and openness.",
+      },
+      {
+        type: "paragraph",
+        content:
+          "One of the primary benefits of open source is the promotion of digital autonomy. By providing access to the source code, these programs empower users to control their own technology, customize software to fit their needs, and avoid vendor lock-in. This level of transparency also allows for greater security, as anyone can inspect the code for vulnerabilities or malicious elements. As a result, users are not solely dependent on a single company for updates, bug fixes, or continued support.",
+      },
+      {
+        type: "paragraph",
+        content:
+          "Additionally, open development fosters innovation and collaboration. Developers can build upon existing projects, share improvements, and learn from each other, accelerating the pace of technological advancement. The open nature of these projects often leads to higher quality software, as bugs are identified and fixed more quickly by a diverse group of contributors. Furthermore, using open source can reduce costs for individuals, businesses, and governments, as it is often available for free and can be tailored to specific requirements without expensive licensing fees.",
+      },
+    ],
+  });
+
+  // Renders the editor instance using a React component.
+  return (
+    <div>
+      <BlockNoteView editor={editor} />
+
+      <div className={"edit-buttons"}>
+        {/*Inserts a new block at start of document.*/}
+        <button
+          className={"edit-button"}
+          onClick={async () => {
+            const blockToChange = editor.document[1].id;
+
+            // Let's get the stream tools so we can invoke them manually
+            // In this case, we're using the default stream tools, which allow all operations
+            const tools = llmFormats.html
+              .getStreamToolsProvider()
+              .getStreamTools(editor, true);
+
+            // Create an executor that can execute StreamToolCalls
+            const executor = new StreamToolExecutor(tools);
+
+            // Use `executeOne` to invoke a single, non-streaming StreamToolCall
+            await executor.executeOne({
+              type: "update",
+              id: blockToChange,
+              block: "<p>Open source software is cool</p>",
+            });
+
+            // make sure the executor is done
+            await executor.waitTillEnd();
```

Comment:
This API is awkward, I'd think it'd be more like:

```js
await executor.createExecutionCtx((execute)=>{
await execute.one({ ...opts })
})
```

So you don't have to remember this `waitTillEnd` thing